pkg/utils/filters/rewrite.go

```diff
@@ -0,0 +1,40 @@
+package filters
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+const OriginPathHeaderKey = "X-Rewrite-Original-Path"
+const OldResourceApiServerPrefixWithoutSlash = "/apis/clusterpedia.io/v1beta1/resources"
+const OldResourceApiServerPrefix = OldResourceApiServerPrefixWithoutSlash + "/"
+
+func WithRewriteFilter(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		oldPath := req.URL.EscapedPath()
+		if rewritePath, ok := urlPrefixRewrite(oldPath); ok {
+			req.URL.Path = rewritePath
+			req.Header.Set(OriginPathHeaderKey, oldPath)
+			klog.V(5).InfoS("request need rewrite", "oldPath", oldPath, "newPath", req.URL.EscapedPath())
+		} else {
+			klog.V(5).InfoS("request not need rewrite", "path", oldPath)
+		}
+
+		handler.ServeHTTP(w, req)
+	})
+}
+
+func urlPrefixRewrite(oldPath string) (string, bool) {
+	if strings.HasPrefix(oldPath, OldResourceApiServerPrefix) || oldPath == OldResourceApiServerPrefixWithoutSlash {
```

Comment:
There are more than just `resources` resource under `clusterpedia.io` group, we can filter out resources that start with `apis/clusterpedia.io`.

Reply:
If we process with prefix `apis/clusterpedia.io`, how to process resouce `/apis/clusterpedia.io/v1beta1/clusters` ?

Reply:
There is no clusters resource in the `clusterpedia.io` group
pediacluster resource in `clusters.clusterpedia.io` group

Reply:
Yes, if not support, it will effect apiserice.